Android/app/src/main/java/io/github/project_travel_mate/destinations/description/FinalCityInfoActivity.java

```diff
@@ -169,6 +173,12 @@ public void onClick(View v) {
                 intent = TweetsActivity.getStartIntent(FinalCityInfoActivity.this, mCity);
                 startActivity(intent);
                 break;
+            case R.id.weather:
+                intent = WeatherActivity.getStartIntent(FinalCityInfoActivity.this, mCity);
+                //pass current temperature to weather activity
+                intent.putExtra(CURRENT_TEMP, mCurrentTemp);
```

Comment:
instead of passing this as extra here. add this parameter to WeatherActivity.getStartIntent() function